locals.tf

```diff
@@ -163,12 +170,14 @@ locals {
     var.distributed_mode ? null : {
       loki-stack = {
         loki = {
-          serviceName = "loki-stack.loki-stack"
+          serviceName = "loki-stack.${var.namespace}"
+          # TODO serviceMonitor is a KPS CRD, manage this circular dependency
           serviceMonitor = {
             enabled = true
           }
         }
         promtail = {
+          # Same previous comment
           serviceMonitor = {
             enabled = true
           }
```

Comment:
I don´t think these are circular dependencies. As long as Loki is deployed after kube-prometheus-stack, there should be no issue, since the latter does not depend on the former (Loki does not deploy any CRDs that I'm aware).

Reply:
Actually Loki is deployed in all setups before KPS otherwise Grafana pod needs to be restarted manually so the sidecar loads Loki datasource. But if this is okay for everyone I'll remove the comment.